kubernetes_asyncio/leaderelection/resourcelock/configmaplock.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+
+from kubernetes_asyncio import client
+from kubernetes_asyncio.client.rest import ApiException
+
+from ..leaderelectionrecord import LeaderElectionRecord
+
+logging.basicConfig(level=logging.INFO)
+
+
+class ConfigMapLock:
```

Comment:
I'd rather use the [Lease API](https://kubernetes.io/docs/concepts/architecture/leases/) instead of this older ConfigMap approach, but the ConfigMap approach is what is currently available in `kubernetes-client/python`.

If the rest of this module looks alright, I can add a leases implementation in a similar style.

Reply:
The author of the open PR in that project has not signed the CLA yet. I've [sought confirmation](https://github.com/kubernetes-client/python/pull/2314#issuecomment-2574097546) of the licensing status of their contribution before attempting to adapt it for inclusion here.

Reply:
As per https://github.com/kubernetes-client/python/pull/2314#issuecomment-2575382136 the CLA has been agreed to.